src/code/providers/provider-interface.ts

```diff
@@ -0,0 +1,550 @@
+// TODO: This file was created by bulk-decaffeinate.
+// Sanity-check the conversion and remove this comment.
+/*
+ * decaffeinate suggestions:
+ * DS201: Simplify complex destructure assignments
+ * DS203: Remove `|| {}` from converted for-own loops
+ * DS205: Consider reworking code to avoid use of IIFEs
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
+
+import isString  from '../utils/is-string'
+import _ from 'lodash'
+
+const FILE_EXTENSION_DELIMETER = "."
+
+type providerType = string
+type parentType = string // Based on the code, seems like this might path?
+type providerDataType = any  // Not sure what this is
+type documentId = string
+type key = string
+type mimeType = string
+
+
+// TODO: What does a save callback signature really looklike?
+type ISaveCallback = (arg: any) => any
```

Comment:
I'd suggest type aliasing the `any` s that you want to eventually tighten up to a concrete type so that you can search for them later as some of the `any` types should stay `any`.  Something like:

```
type IAnyForNow = any
type ISaveCallback = (arg: IAnyForNow) => IAnyForNow
```

Reply:
If you put that `IAnyForNow` in an importable module then you can use VS Code's "find all uses" command.

Reply:
One other thing - in other projects we have been reserving the `I` prefix for interfaces and not using it on types.